src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,84 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import classNames from 'classnames';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  activeTodo: Todo;
+  setActiveTodo: (todo: Todo | null) => void;
+  setShowModal?: (value: boolean) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  activeTodo,
+  setActiveTodo = () => {},
+  setShowModal = () => {},
```

Comment:
Initializing state with props can lead to issues where the local state does not reflect changes in the props if the props change. Consider using the effect hook to sync state with props when necessary. More info here: https://stackoverflow.com/a/50403930